docs/design/2018-11-29-ast-to-sql-text.md

```diff
@@ -0,0 +1,143 @@
+# Proposal: Support to restore SQL text from a AST tree.
+
+- Author(s):     [Yilin Zhao](https://github.com/leoppro)
+- Last updated:  2018-11-29
+- Discussion at: https://github.com/pingcap/tidb/issues/8532
+
+## Abstract
+
+This proposal aims to support to restore SQL text from any `ast.Node`.
+
+## Background
+
+We know there is a `Text()` function of `ast.Node`, 
+the parser calls `SetText()` function during the parsing process, then we can call `Text()` to get original input sql. 
+But the `Text()` function is incomplete, only the `Text()` of root nodes can work well. 
+
+Now some features(eg. Database name and table name mapping) need us to make any `ast.Node` could restore to SQL text.
+
+## Proposal
+
+We know the AST is a type of tree structure of which child node corresponding to SQL text. 
+So we can walk through AST tree layer by layer and splice SQL text according to AST node. 
+There is a AST node parsed from `SELECT column0 FROM table0 UNION SELECT column1 FROM table1 WHERE a = 1`, 
+we can splice SQL text as the picture shows.
+
+![ast tree](./imgs/ast-tree.png)
+
+## Rationale
+
+See: [Rationale (Chinese)](https://docs.google.com/document/d/1KF4OvObvAzT0YW_KwiMRrsGZQMNVE03RdqShhStM_D4/edit?usp=sharing)
```

Comment:
Do not link the Chinese doc here.
Please translate the rationale into English.

Reply:
i remove it, i think the doc is not helpful to understanding this proposal